cni-plugin/linkerd-cni-validation/src/main.rs

```diff
@@ -0,0 +1,170 @@
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+use anyhow::{anyhow, Context, Result};
+use clap::Parser;
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream},
+};
+use tracing::{debug, error, info, Instrument};
+use tracing_subscriber::prelude::*;
+use tracing_subscriber::EnvFilter;
+
+#[derive(Parser)]
+#[clap(version)]
+struct Args {
+    #[clap(long, default_value = "4140")]
+    outbound_port: u16,
+    #[clap(
+        parse(try_from_str),
+        long,
+        env = "LINKERD_VALIDATION_LOG_LEVEL",
+        default_value = "debug"
+    )]
+    log_level: EnvFilter,
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_HOST")]
+    target_host: Ipv4Addr,
+    #[clap(parse(try_from_str), long, env = "KUBERNETES_SERVICE_PORT")]
+    target_port: u16,
+}
+
+static REDIRECT_RESPONSE: &str = "REDIRECTION SUCCESSFUL";
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let Args {
+        outbound_port,
+        log_level,
+        target_host,
+        target_port,
+    } = Args::parse();
+
+    tracing_subscriber::registry()
+        .with(log_level)
+        .with(tracing_subscriber::fmt::layer())
+        .try_init()?;
+
+    let target_addr = &format!("{}:{}", target_host, target_port);
+    info!(original_dst = %target_addr, outbound_redirect = %outbound_port, "validating outbound traffic is redirected to proxy outbound port");
+
+    let (shutdown_tx, shutdown_rx) = drain::channel();
+    let (ready_tx, ready_rx) = tokio::sync::oneshot::channel::<Result<()>>();
+
+    let srv_handle = tokio::spawn(async move {
+        let listen_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), outbound_port);
```

Comment:
```suggestion
        let listen_addr = SocketAddr::new([0, 0, 0, 0].into(), outbound_port);
```

but I think we can avoid pinning this to IPv4 if we take an address on the CLI